### DISCUSSION TITLE
[ASK] Can stern used as golang library?

### DISCUSSION THREAD

**Discussion by kholisrag** [stern/stern#184] [Q&A]
Hello All,
Can stern used as golang library?
basically I want to create tools to watch and stream kubernetes logs, and do something if match,

is this tool can be used as package / library directly in golang code?,
can anyone volunteer to provide simple example using this as library and stream / watch the logs?

Thanks for your help!

**Comment by superbrothers** [comment]
Yes, it can. However, stern is not intended to be used as a library. As a result, the interface and behavior can change significantly. Please note that.
